JsonE/Expressions/ArrayExpressionNode.cs

```diff
@@ -57,7 +58,7 @@ public bool TryParse(ReadOnlySpan<char> source, ref int index, out ExpressionNod
 			if (!ExpressionParser.TryParse(source, ref i, out var value))
 				throw new SyntaxException(CommonErrors.WrongToken(source[i]));
 
-			arr.Add(JsonExpression.Create(value!));
+			arr.Add((JsonNode)(JsonValue)JsonExpression.Create(value!));
```

Comment:
What's going on here?  Double casts are smelly.

Reply:
`arr.Add(JsonNode?)` is AOT safe, but `arr.Add<T>(T)` is not. This makes sure we go through the right one.

I'm open to suggestions of the right way to do this.

Reply:
Oh, I didn't even notice it was using a generic!

Maybe just put it in a variable first?
```suggestion
			JsonNode expr = JsonExpression.Create(value!);
			arr.Add(expr);
```

How does that sit?

Reply:
Yeah, if that compiles I'll take it. :)